cm/list_test.go

```diff
@@ -14,3 +16,53 @@ func TestListMethods(t *testing.T) {
 		t.Errorf("got (%s) != want (%s)", string(got), string(want))
 	}
 }
+
+func TestListJSON(t *testing.T) {
```

Comment:
Can you split this up into two tests, each with a list of test cases? e.g. `TestListMarshalJSON` and `TestListUnmarshalJSON`? Something like:

```go
func TestListMarshalJSON(t *testing.T) {
	tests := []struct {
		name    string
		v       json.Marshaler
		want    string
		wantErr bool
	}{
		// ...
	}
	// ...
}
```

Reply:
sure thing. what other test cases would you like to see there?

Reply:
Various examples of lists, with different types (including structs), empty lists, etc. We can have this test suite adhere to (or inform) a WIT to JSON spec.